nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -314,6 +258,34 @@ func getHostCredential(m interface{}) (string, string) {
 	return username, password
 }
 
+func getMatchingIPVersion(ip string, hostIPs []string) string {
+	needIPv4 := (net.ParseIP(ip)).To4() != nil
+
+	for _, hostIP := range hostIPs {
+		isIPv4 := (net.ParseIP(hostIP)).To4() != nil
+		if needIPv4 == isIPv4 {
+			return hostIP
+		}
+	}
+
+	return ""
+}
+
+func isMatchingNode(node nsxModel.ClusterNodeInfo, address string) bool {
+	for _, entity := range node.Entities {
+		if entity.Port != nil {
+			if (entity.IpAddress != nil) && (*entity.IpAddress == address) {
+				return true
+			}
+			if (entity.Ipv6Address != nil) && (*entity.Ipv6Address == address) {
```

Comment:
shall we convert string to IPs to avoid equality check to fail due to differences in Ipv6 short/full representation?